icepyx/core/spatial.py

```diff
@@ -510,3 +599,86 @@ def extent_file(self):
         ./doc/source/example_notebooks/supporting_files/simple_test_poly.gpkg
         """
         return self._geom_file
+
+    # ----------------------------------------------------------------------
+    # Methods
+
+    # TODO: can use this docstring as a todo list
+    def fmt_for_CMR(self):
+        """
+        Format the spatial extent for the CMR API.
+
+        CMR spatial inputs must be formatted a specific way.
+        This method formats the given spatial extent to be a valid submission.
+        For large/complex polygons, this includes simplifying the polygon (NOTE: currently not all polygons are simplified enough).
+        Coordinates will be properly ordered, and the required string formatting applied.
+        For small regions, a buffer may be added.
+
+        Returns
+        -------
+        string
+            Properly formatted string of spatial data for submission to CMR API.
+
+
+        """
+        # CMR keywords: ['bounding_box', 'polygon']
+        if self._ext_type == "bounding_box":
+            cmr_extent = ",".join(map(str, self._spatial_ext))
+
+        elif self._ext_type == "polygon":
+            poly = self.extent_as_gdf.geometry
+
+            if any(
+                geomtype in ["MultiPoint", "MultiLineString", "MultiPolygon"]
+                for geomtype in poly.geom_type
+            ):
+                poly = poly.convex_hull
+
+            poly = poly.unary_union
+
+            # Simplify polygon. The larger the tolerance value, the more simplified the polygon. See Bruce Wallin's function to do this
+            poly = poly.simplify(0.05, preserve_topology=False)
```

Comment:
Is this [Douglas-Peucker polygon simplification algorithm](https://shapely.readthedocs.io/en/1.8.5.post1/manual.html#object.simplify) for #159? I realize that getting an error is not nice, but am worried about unintended consequences of simplifying the polygon on behalf of the user. The tolerance of [0.05deg is about 5.6km](https://www.wolframalpha.com/input/?i=0.05deg+to+metres) which is not a small distance.

Reply:
> Is this [Douglas-Peucker polygon simplification algorithm](https://shapely.readthedocs.io/en/1.8.5.post1/manual.html#object.simplify) for addressing #159?

Not quite. There are multiple polygon submission issues going on, and this PR does not address #159. This function is only for the CMR step, i.e. during potential granule identification to get the user some info about available data. With cloud access, since we're using this step to actually get granule IDs, we could now be introducing a problem, but for on-prem data order/download, these results are never used and the polygon is not simplified when it's submitted in that step (and this is where #159 comes in). The SlideRule team has been working hard on these issues (there are also issues with the fact that metadata and actual content don't always align for each granlue, especially during off-pointing). After we've switched over to using earthaccess (currently still called earthdata), which is where this type of functionality (simplifying, buffering, etc.) should probably ultimately live anyway, I plan to implement some of the solutions SR has developed to handle some of these tiny and complex polygon issues.

For right now, this formatting is just being moved from APIformatting into the spatial module (and generalized/fixed to handle cross dateline issues).